### PR TITLE
Added startsWith and endsWith assertions

### PR DESCRIPTION
**Pull request by ILikeYourHat**
Added `startsWith` and `endsWith` for `List` assertions.

This is my first contribution, so hello everybody 👋

### CODE REVIEW

File: assertk/src/commonMain/kotlin/assertk/assertions/list.kt [hunk 1]
```diff
@@ -94,4 +94,64 @@ fun Assert<List<*>>.containsSubList(sublist: List<*>) = given { actual: List<*>
...
+    val sublist = if (actual.size >= elements.size) {
+        actual.subList(actual.size - elements.size, actual.size)
+    } else {
+        actual
+    }
+
+    if (sublist.contentEquals(elements)) return
+
+    expected(
+        "to end with:${
```
Comment (evant) [inline assertk/src/commonMain/kotlin/assertk/assertions/list.kt:151]: Hm, the other assertions will show the entire list for context which I think is helpful. I think wouldn't be a problem to do that with `startsWith` but might be hard to read with `endsWith`, maybe showing the sublist in addition if necessary?

```kotlin
if (isSublist) {
  expected("to end with:${show(elements)} but was ${show(actual)}\n ending not expected:${show(sublist)}")
} else {
  expected("to end with:${show(elements)} but was ${show(actual)}")
}
```
you can see similar assertions in iterable.kt.

Thoughts?

**Review by evant — CHANGES_REQUESTED** [review]

File: assertk/src/commonMain/kotlin/assertk/assertions/list.kt (hunk 1 above)
Comment (ILikeYourHat) [reply assertk/src/commonMain/kotlin/assertk/assertions/list.kt:151]: Maybe `expected("to end with:${show(elements)} but was ${show(sublist)} in ${show(actual)}` for both cases? This is some duplication, but you have sublist at the beginning to verify what's wrong, and full list at the end for context

Comment (evant) [reply assertk/src/commonMain/kotlin/assertk/assertions/list.kt:151]: Sounds good

Comment (ILikeYourHat) [reply assertk/src/commonMain/kotlin/assertk/assertions/list.kt:153]: changed

**Review by evant — APPROVED** [review]

**Pull request merged**